datahub-web-react/src/app/entity/shared/components/styled/search/action/DeprecationDropdown.tsx

```diff
@@ -1,27 +1,80 @@
-import React from 'react';
+import { message, Modal } from 'antd';
+import React, { useState } from 'react';
+import { useBatchUpdateDeprecationMutation } from '../../../../../../../graphql/mutations.generated';
+import { UpdateDeprecationModal } from '../../../../EntityDropdown/UpdateDeprecationModal';
 import ActionDropdown from './ActionDropdown';
 
 type Props = {
     urns: Array<string>;
     disabled: boolean;
+    refetch?: () => void;
 };
 
 // eslint-disable-next-line
-export default function DeprecationDropdown({ urns, disabled = false }: Props) {
-    return (
-        <ActionDropdown
-            name="Deprecation"
-            actions={[
-                {
-                    title: 'Mark as deprecated',
-                    onClick: () => null,
-                },
-                {
-                    title: 'Mark as undeprecated',
-                    onClick: () => null,
+export default function DeprecationDropdown({ urns, disabled = false, refetch }: Props) {
+    const [isEditModalVisible, setIsEditModalVisible] = useState(false);
+    const [batchUpdateDeprecationMutation] = useBatchUpdateDeprecationMutation();
+
+    const batchUndeprecate = () => {
+        batchUpdateDeprecationMutation({
+            variables: {
+                input: {
+                    resources: [...urns.map((urn) => ({ resourceUrn: urn }))],
+                    deprecated: false,
                 },
-            ]}
-            disabled={disabled}
-        />
+            },
+        })
+            .then(({ errors }) => {
+                if (!errors) {
```

Comment:
nit: I think if there are errors then it should skip this block and go straight to the `catch` block?

Reply:
Yeah tbh im quite confused abt this